core-rust/state-manager/src/metrics.rs

```diff
@@ -330,6 +344,202 @@ impl CommittedTransactionsMetrics {
     }
 }
 
+impl ProtocolMetrics {
+    pub fn new(registry: &Registry, initial_protocol_state: &ProtocolState) -> Self {
+        let instance = Self {
+            protocol_update_readiness_ratio: GaugeVec::new(
+                opts(
+                    "protocol_update_readiness_ratio",
+                    "A ratio of supporting stake to total stake in the current validator set.",
+                ),
+                &["readiness_signal_name"],
+            )
+            .registered_at(registry),
+            pending_update_threshold_required_ratio: GaugeVec::new(
+                opts(
+                    "protocol_update_pending_threshold_required_ratio",
+                    "Required readiness ratio for the given protocol update threshold.",
+                ),
+                &["protocol_version_name", "readiness_signal_name", "threshold_index"],
+            )
+                .registered_at(registry),
+            pending_update_threshold_required_consecutive_epochs: IntGaugeVec::new(
+                opts(
+                    "protocol_update_pending_threshold_required_consecutive_epochs",
+                    "Required number of consecutive epochs of support for the given protocol update threshold.",
+                ),
+                &["protocol_version_name", "readiness_signal_name", "threshold_index"],
+            )
+                .registered_at(registry),
+            pending_update_threshold_matching_consecutive_epochs: IntGaugeVec::new(
+                opts(
+                    "protocol_update_pending_threshold_matching_consecutive_epochs",
+                    "Current number of consecutive epochs of support for the given protocol update threshold.",
+                ),
+                &["protocol_version_name", "readiness_signal_name", "threshold_index"],
+            )
+                .registered_at(registry),
+            pending_update_lower_bound_epoch: IntGaugeVec::new(
+                opts(
+                    "protocol_update_pending_lower_bound_epoch",
+                    "Earliest epoch when the given protocol update can be enacted (inclusive)",
+                ),
+                &["protocol_version_name", "readiness_signal_name"],
+            )
+                .registered_at(registry),
+            pending_update_upper_bound_epoch: IntGaugeVec::new(
+                opts(
+                    "protocol_update_pending_upper_bound_epoch",
+                    "Upper bound epoch for the given protocol update (exclusive)",
+                ),
+                &["protocol_version_name", "readiness_signal_name"],
+            )
+                .registered_at(registry),
+            enacted_protocol_update_state_version: IntGaugeVec::new(
+                opts(
+                    "protocol_update_enacted_state_version",
+                    "State version of enacted protocol update",
```

Comment:
Maybe worth clarifying it's the state version of the init of the protocol update?